libs/js-sdk/src/client.ts

```diff
@@ -68,30 +72,31 @@ export function createBrowserClient({
     updateStrategyImplementation
         .connect(state)
         .then(() => {
-            if (!initialisedPromise.completed) {
+            if (initialPromise !== initialisedState.initialisedPromise) {
```

Comment:
This allows us to throw away 'old' in flight responses.

This code probably could be rewritten to use cancellation tokens to cancel the previous in flight request but that's a larger change than just checking object equality to see if the promise has changed